pkg/gadgets/run/types/metadata_test.go

```diff
@@ -158,6 +158,48 @@ func TestValidate(t *testing.T) {
 				},
 			},
 		},
+		"toppers_more_than_one": {
+			objectPath: "../../../../testdata/validate_metadata1.o",
+			metadata: &GadgetMetadata{
+				Name: "foo",
+				Toppers: map[string]Topper{
+					"foo": {},
+					"bar": {},
+				},
+			},
+			expectedErrString: "only one topper is allowed",
+		},
+		"toppers_bad_map_type": {
+			objectPath: "../../../../testdata/validate_metadata1.o",
+			metadata: &GadgetMetadata{
+				Name: "foo",
+				Toppers: map[string]Topper{
+					"foo": {
+						MapName:    "events",
+						StructName: "event",
+					},
+				},
+				Structs: map[string]Struct{
+					"event": {},
+				},
+			},
+			expectedErrString: "map \"events\" has a wrong type: expected: hash",
+		},
+		"toppers_good": {
+			objectPath: "../../../../testdata/validate_metadata1.o",
+			metadata: &GadgetMetadata{
+				Name: "foo",
+				Toppers: map[string]Topper{
+					"foo": {
+						MapName:    "myhashmap",
+						StructName: "event",
```

Comment:
> IMO, we shouldn't support non-BTF maps for toppers and support this check instead. Additionally, it would align things between using the GADGET_TOPPER macro and providing a manually created metadata file.

I agree, we shouldn't support non-BTF maps.

Reply:
Was this check implemented? I meant to have something like 

(EDIT: I based this on an older branch, so it doesn't apply cleanly, but the idea is the same)

```diff
diff --git pkg/gadgets/run/types/metadata.go pkg/gadgets/run/types/metadata.go
index c2aac7572..4d50d892f 100644
--- pkg/gadgets/run/types/metadata.go
+++ pkg/gadgets/run/types/metadata.go
@@ -262,6 +262,29 @@ func validateTracerMapType(mapType ebpf.MapType) error {
        return nil
 }
 
+func (m *GadgetMetadata) validateTopper(t Topper, name string, spec *ebpf.CollectionSpec) error {
+       err := validateMapAndStruct(t.MapName, t.StructName, spec, m, validateTopperMapType)
+       if err != nil {
+               return fmt.Errorf("validating topper %q: %w", name, err)
+       }
+
+       topperMap := spec.Maps[t.MapName]
+       if topperMap.Value == nil {
+               return fmt.Errorf("map %q does not have BTF information for its values", topperMap.Name)
+       }
+       topperMapStruct, ok := topperMap.Value.(*btf.Struct)
+       if !ok {
+               return fmt.Errorf("map %q's value is not a structure", topperMap.Name)
+       }
+
+       if topperMapStruct.Name != t.StructName {
+               return fmt.Errorf("map %q's value struct (%q) and struct name (%q) don't match",
+                       topperMap.Name, topperMapStruct.Name, t.StructName)
+       }
+
+       return nil
+}
+
 func (m *GadgetMetadata) validateToppers(spec *ebpf.CollectionSpec) error {
        var result error
 
@@ -271,10 +294,7 @@ func (m *GadgetMetadata) validateToppers(spec *ebpf.CollectionSpec) error {
        }
 
        for name, t := range m.Toppers {
-               err := validateMapAndStruct(t.MapName, t.StructName, spec, m, validateTopperMapType)
-               if err != nil {
-                       result = multierror.Append(result, fmt.Errorf("validating topper %q: %w", name, err))
-               }
+               result = multierror.Append(result, m.validateTopper(t, name, spec))
        }
 
        return result
```